Story/RainMeadow.StoryHooks.cs

```diff
@@ -171,9 +180,71 @@ private bool AllPlayersThroughtoOtherSide(On.RegionGate.orig_AllPlayersThroughTo
                 }
             }
             return true;
-
             // Do not call original method. 
-            // TODO: Can we access abstractCreature pos through .apo safely?
+        }
+
+
+        public int PlayersInZone(On.RegionGate.orig_PlayersInZone orig, RegionGate self)
+        {
+
+            if (OnlineManager.lobby == null || OnlineManager.lobby.gameMode is not StoryGameMode)
+            {
+                orig(self);
+                return -1;
+
+            }
+
+            int num = -1;
+            foreach (var playerAvatar in OnlineManager.lobby.playerAvatars)
+            {
+
+                int num2 = self.DetectZone(playerAvatar.Value); // wants an AbstractCreature
```

Comment:
@Chrometurtle, DetectZone wants a parameter of type AbstractCreature. Even after an attempted cast, the compiler warns that playerAvatar.Value is of type OnlineCreature. Wanted to pick your brain on that.

Reply:
playerAvatar.Value.realizedCreature.abstractCreature

Just keeping records for people to see if they're interested

Reply:
can probably be simplified to `(playerAvater.Value.apo as AbstractCreature)`